app/controllers/miq_request_controller.rb

```diff
@@ -400,6 +400,17 @@ def states_for_checkboxes_i18n
     end
   end
 
+  def user_options(params)
+    set_defaults = prov_set_default_options
+    user_defaults = {
```

Comment:
This assignment is also now unnecessary.

Reply:
since we're crunching it down to 
```
  def user_options(params)
    user_defaults = {
      :reason_text    => params["reasonText"] || prov_set_default_options[:reason_text],
      :applied_states => params["approvalStateCheckboxes"] || prov_set_default_options[:applied_states],
      :type_choice    => params["types"] || prov_set_default_options[:type_choice],
      :user_choice    => params["selectedUser"] || prov_set_default_options[:user_choice],
      :time_period    => params["selectedPeriod"] || prov_set_default_options[:time_period],
    }
  end
```
should we take it a step further and make it
```
  def user_options(params)
    user_defaults = {
      :reason_text    => params["reasonText"] || nil,
      :applied_states => params["approvalStateCheckboxes"] || PROV_STATES.keys,
      :type_choice    => params["types"] || 'all',
      :user_choice    => params["selectedUser"] || (approver? ? 'all' : current_user.id),
      :time_period    => params["selectedPeriod"] || 7,
    }
  end
```
removing the need for `def prov_set_default_options`

Reply:
Don't think so as `prov_set_default_options` is also used here: https://github.com/ManageIQ/manageiq-ui-classic/blob/5d134b2ea12b728f19bdd68b3f642985f30e0745/app/controllers/miq_request_controller.rb#L82

Reply:
also, you should be crunching it down to:
```ruby
def user_options(params)
    {
      :reason_text    => params["reasonText"] || prov_set_default_options[:reason_text],
      :applied_states => params["approvalStateCheckboxes"] || prov_set_default_options[:applied_states],
      :type_choice    => params["types"] || prov_set_default_options[:type_choice],
      :user_choice    => params["selectedUser"] || prov_set_default_options[:user_choice],
      :time_period    => params["selectedPeriod"] || prov_set_default_options[:time_period],
    }
  end
```

Reply:
That line was updated to be `@view, @pages = get_view(kls, :named_scope => prov_scope(user_options(params)))` see [here](https://github.com/ManageIQ/manageiq-ui-classic/pull/7861/files#diff-924590069c6f62d0466390965d8484f465114e0c74d24de8bf441f535c549ed9R82), so `def user_options(params)` is the only place that uses `prov_set_default_options`

Reply:
@Fryguy is it worthwhile to keep around a default_options method with only a single caller?

Reply:
@Fryguy notification bump

Reply:
Probably not unless it's part of a more general pattern across the ui